### DISCUSSION TITLE
Workers healthcheck/uptime

### DISCUSSION THREAD

**Discussion by andresbarcenas** [ruby-amqp/kicks#10] [General]
Hello,

We have over 100+ workers and we are seeing random workers crashing but we have no logs nor we can tell why its happening. Is there a way to log a heartbeat or something similar to tell if a worker is not responding? Our normal reaction is to restart the entire sneakers process

**Comment by simi** [comment]
Hard to judge just by the info provided. Per my experience mostly OOM killer is the reason for unexpected kills. But it heavily depends on environment.

**Comment by michaelklishin** [comment]
OOM killer events are logged in one of the kernel/OS log files.

We won't guess as to what may be happening, the very first step of this investigation should be capturing Kicks worker process logs. How to do that and what may be affecting, redirecting or collecting the logs, only you know and is not up to the community to try to figure out.
